### DISCUSSION TITLE
AGH blocks higher level domain unexpectedly.

### DISCUSSION THREAD

**Discussion by Kainichy** [AdguardTeam/AdGuardHome#2812] [Q&A] [answered]
### Issue Details

* **Version of AdGuard Home server:**
  * v0.105.2-beta.1.
* **How did you install AdGuard Home:**
  * Downloaded from the [link](https://static.adguard.com/adguardhome/beta/AdGuardHome_linux_arm64.tar.gz) of Beta version from README.md.
* **How did you setup DNS configuration:**
  * IoT.
* **Device model:**
  * Raspberry Pi 4 Model B.
* **CPU architecture:**
  * ARM.
* **Operating system and version:**
  * Raspbian GNU/Linux 10 (buster) aarch64
  * `uname -a`: `Linux <Hostname> 5.10.20-v8+ #1404 SMP PREEMPT Thu Mar 4 19:48:26 GMT 2021 aarch64 GNU/Linux`

### Expected Behavior
With `127.0.0.1 minorshort.weixin.qq.com` in my DNS blocklist, `weixin.qq.com` should be resolved.

### Actual Behavior
With that rule above, AdGuard Home doesn't resolve `weixin.qq.com`.

### Screenshots

<details><summary>Screenshot:</summary>
<img width="855" alt="Screen Shot 2021-03-14 at 19 33 19" src="https://user-images.githubusercontent.com/44992049/111066954-b64c2280-84fc-11eb-84b7-81d4fc3214db.png">
<img width="502" alt="Screen Shot 2021-03-14 at 19 33 10" src="https://user-images.githubusercontent.com/44992049/111066967-c3691180-84fc-11eb-9149-58790132e11a.png">
</details>

**Comment by ameshkov** [comment, marked as answer]
Actually, this is not a bug.

AdGuard Home matches CNAME record (`weixin.qq.com` points to `minorshort.weixin.qq.com`).
If you want it to not touch `weixin.qq.com`, add this rule: `@@|weixin.qq.com^`

**Reply by Kainichy** [reply]
Got it. Thanks.
Creating issue for the blocklist...
